lusid-sdk-python/tests/business_agility_test_environment.py

```diff
@@ -0,0 +1,662 @@
+                'quantity': 8954300,
+                'price': 4.56900,
+                'currency': 'GBP'}
+        }
+
+    @timeit
+    def create_portfolio_production_environment(self):
+        """
+        The first thing that we need to do is create a production environment with an empty transaction portfolio which
+        we can use to hold the assets of the pension fund.
+        We will test that the production environment and pension fund portfolio are created successfully
+        """
+
+        # Create our pension fund portfolio in the production scope
+        pension_fund_request = models.CreateTransactionPortfolioRequest(display_name='pension-fund',
+                                                                        code=self.portfolio_code,
+                                                                        base_currency='GBP',
+                                                                        description='Pension fund for new client')
+
+        pension_fund_portfolio = self.client.create_portfolio(scope=self.production_scope_code,
+                                                              create_request=pension_fund_request)
+
+        # Test that our portfolio was created successfully
+        self.portfolio_creation_tests(pension_fund_portfolio, pension_fund_request, self.production_scope_code)
+
+    @timeit
+    def add_holdings_portfolio_production_environment(self):
+        """
+        Now that we have created our portfolio in the production environemnt, we need to take the final holdings report
+        that we received from the transition manager and populate our portfolio.
+        Before we do this, we first need to do add the instruments to LUSID. Once we have added the instruments we can
+        then populate our initial holdings on these instruments.
+        We will test that we have correctly created the instruments and added the holdings to our portfolio
+        """
+
+        '''
+        Rather than using separate update and insert methods, LUSID uses an upsert method for may of its operations.
+
+        This inserts a record if it does not already exist and modifies an existing record if it does already exist.
+
+        We can upsert our instruments individually or in a batch. Either way the request should be in the form of a
+        dictionary with an arbitrary reference name for the instrument as the key and an InstrumentDefinition object
+        as the value.
+
+        The InstrumentDefinition requires a name for the instrument which is a string. In this case we will use the 
+        same name as the arbitrary reference key.
+
+        It also requires a dictionary containing at least one unique identifier for the asset. The key for this 
+        dictionary is the name of the identifier, in this case we will use 'ClientInternal'. The value is a string which
+        is the identifier. The allowed identifiers currently supported are:
+
+            - ClientInternal
```

Comment:
can this be replaced to a link to the appropriate API call on docs.lusid.com so it doesn't become out of date (it already is)

Reply:
Linked to https://docs.lusid.com/#operation/GetInstrumentIdentifiers